File: LeetCode/JAVA/LeetCode206.java
public class LeetCode206 {

    public static class Node{
        int data;
        Node next;
        Node(int d){
            data = d;
            next = null;
        }
    }

    public static Node head;

    public static Node reverseLL(Node head){
        Node prev = null;
        Node curr = head;
        Node nextNode;
        while(curr!=null){
            nextNode = curr.next;
            curr.next = prev;
            prev = curr;
            curr = nextNode;
        }
        return prev;
    }

    public static void main(String[] args) {
        Node newhead = new Node(1);
        newhead.next = new Node(2);
        newhead.next.next = new Node(3);
        newhead.next.next.next = new Node(4);
        newhead.next.next.next.next = new Node(5);

        Node res = reverseLL(newhead);

        System.out.print("[");
        while(res!=null){
            System.out.print(res.data);
            res=res.next;
            if(res!=null)
            System.out.print(", ");
        }
        System.out.print("]");
    }

}

File: LeetCode/CPP/LeetCode206.cpp
#include<iostream>
#include<vector>

using namespace std;

struct Node{
    int data;
    Node* next;
    Node(int datad){
        data = datad;
        next = nullptr;
    }
};

Node* head = nullptr;

Node* reverseLinkedList(Node* head){
    Node* prev = nullptr;
    Node* curr = head;
    Node* next = nullptr;
    while (curr!=nullptr)
    {
        next = curr->next;
        curr->next = prev;
        prev = curr;
        curr = next;
    }
    return prev;
};

int main(){
    Node* newNode = new Node(1);
    newNode->next = new Node(2);
    newNode->next->next = new Node(3);
    newNode->next->next->next = new Node(4);
    newNode->next->next->next->next = new Node(5);
    Node* newHead = reverseLinkedList(newNode);
    cout<<"[";
    while(newHead!=nullptr){
        cout<<newHead->data;
        newHead = newHead->next;
        if(newHead!=nullptr)
        cout<<", ";
    }
    cout<<"]";
    return 0;
}